packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -372,6 +385,33 @@ class _MenuAnchorState extends State<MenuAnchor> {
       }
     }
     assert(_menuController._anchor == this);
+    if (oldWidget.style?.sizeAnimationStyle != widget.style?.sizeAnimationStyle) {
+      updateAnimation();
+    }
+  }
+
+  void updateAnimation() {
+    final AnimationStyle defaultAnimationStyle = _MenuDefaultsM3(context).sizeAnimationStyle;
+    final AnimationStyle effectiveAnimationStyle = widget.style?.sizeAnimationStyle
+      ?? MenuTheme.of(context).style?.sizeAnimationStyle
+      ?? defaultAnimationStyle;
+
+    _animateController.duration = effectiveAnimationStyle.duration;
+    _animateController.reverseDuration = effectiveAnimationStyle.reverseDuration;
+
+    _menuSize = CurvedAnimation(
+      parent: _animateController,
+      curve: effectiveAnimationStyle.curve ?? Curves.linear,
+      reverseCurve: effectiveAnimationStyle.reverseCurve,
+    );
+
+    if (effectiveAnimationStyle == defaultAnimationStyle) {
```

Comment:
Why is this special treatment needed? Does it make custom animation styles less capable?